### ISSUE TITLE
The GHC backend generates incorrect code for Agda code that uses reflection

### ISSUE THREAD

**Issue by nad** [agda/agda#5421]
If you compile the following code using the GHC backend and run the resulting binary, then the result is a crash:
```agda
open import Agda.Builtin.IO
open import Agda.Builtin.Reflection
open import Agda.Builtin.Strict
open import Agda.Builtin.String
open import Agda.Builtin.Unit

postulate
  putStrLn : String → IO ⊤

{-# FOREIGN GHC import qualified Data.Text.IO #-}
{-# COMPILE GHC putStrLn = Data.Text.IO.putStrLn #-}

main : IO ⊤
main = primForce (bindTC {A = Set} {B = Set}) λ _ → putStrLn "Success"
```
```
Bug: MAlonzo Runtime Error: postulate evaluated: Agda.Builtin.Reflection.bindTC
CallStack (from HasCallStack):
  error, called at ./MAlonzo/Code/Agda/Builtin/Reflection.hs:178:5 in main:MAlonzo.Code.Agda.Builtin.Reflection
```
I acknowledge that the code above is not idiomatic, but I don't think that the GHC backend should generate code that crashes in this way.

**Comment by asr** [comment]
What about of cherry-picking the fix into `release-2.6.2`?

**Comment by nad** [comment]
I have no strong opinion on this.
